org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/workspace/internal/DefaultModelProvider.java

```diff
@@ -155,7 +153,12 @@ public T call() {
             U result = (U) this.cache.get(cacheKey, cacheValueLoader);
             return result;
         } catch (Exception e) {
-            throw new GradlePluginsRuntimeException(e);
+            if (e instanceof UncheckedExecutionException && e.getCause() instanceof RuntimeException) {
```

Comment:
I don't think this needs further explanation, it's just the general pattern when you use a LoadingCache.